### DISCUSSION TITLE
aws-rds: How to enable final snapshot for Aurora Cluster

### DISCUSSION THREAD

**Discussion by MetadelDes** [aws/aws-cdk#24622] [General]
### Describe the bug

I am creating database cluster, and i want to enable final snapshot. After creation of the rds, when i navigate to the console, upon modifying the deletion protection and delete the cluster, i dont see the option for "create final snapshot"

here is the code: 
```
.cluster = new DatabaseCluster(this, 'Cluster', {
      engine: rds.DatabaseClusterEngine.auroraPostgres(this.newProps),
      credentials: auroraClusterCrendentials,
      cloudwatchLogsRetention: logs.RetentionDays.ONE_YEAR,
      clusterIdentifier:" namee",
      instanceProps:
      {
        instanceType: props.instanceType,
        vpcSubnets: {
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
        },
        vpc: existingVpc,                  
      },
      parameterGroup: clusterParameterGroup,
      iamAuthentication: true,
      deletionProtection: deletionProtection,
    });
```

### Expected Behavior

I wanted to see create final snapshot selected, is there a way to enable create final snapshot for rds cluster 
![image](https://user-images.githubusercontent.com/74792462/224454321-fe0817fe-b4e3-41b9-b877-93fbab20df24.png)

### Current Behavior

![image](https://user-images.githubusercontent.com/74792462/224455918-37f3e85e-86c7-4f2a-842a-894862cd0caf.png)

### Reproduction Steps

1) Create cluster with above code through cdk,

2) cdk deploy

3) navigate to the console and  modify the cluster to turn off delete protection 

4) try to delete either the reader or writer instance 

the above screen shows up instead of seeing final snapshot option. Could you please tell me how to enable final snapshot option through cdk. I couldn't find info on that. Thanks

### CDK CLI Version

2.44

### Node.js Version

10.17.27

### OS

Microsoft Windows 10 enterprise

### Language

Typescript

### Language Version

3.9.7

**Comment by pahud** [comment]
Hi

Have you tried [removalPolicy.SNAPSHOT](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseCluster.html#removalpolicy) ? When this specified, when you destroy the cluster with CDK, the latest snapshot of the database cluster should be taken.

**Comment by MetadelDes** [comment]
> Hi
> 
> Have you tried [removalPolicy.SNAPSHOT](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseCluster.html#removalpolicy) ? When this specified, when you destroy the cluster with CDK, the latest snapshot of the database cluster should be taken.

@pahud Okay I will try that now. Thanks! One more question how do i set the name of the cluster parameter group and instance parameter group? I dont see a property to set the name from CDK documentation.  I dot want to see the auto generated string. Could you please help?

**Comment by pahud** [comment]
@MetadelDes 


cluster parameter should be https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseCluster.html#parametergroup and instance parameter should be defined in `instanceProps` https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.InstanceProps.html#parametergroup

Please note I haven't verified it in my account yet but according to the doc it should work.

**Comment by pahud** [comment]
I am converting this issue to discussion for general guidance.
